### PR TITLE
Added blog post about jira/github

### PR DESCRIPTION
**Pull request by Fgerthoffert**

### CODE REVIEW

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+Over the past year, we’ve been moving most of our projects away from Jira, in favor of GitHub issues. This blog post will go over the reasons why and lessons we learnt along the way.
+
+It all started with an opportunity. As we were launching an entirely new project [Kids-First](https://kidsfirstdrc.org/) our teams started discussions around the project management tool to be used during implementation. Some of our objectives, in looking at new approaches, were to see how to reinforce integration between stories and actual software code, make developers’ life easier, while still providing management with adequate planning and reporting features.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: , Kids-First,

Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: i would make this one an and instead of ```,```.

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/jira-github/jira-workflow.png" />
+    <figcaption>Workflows can get complicated !</figcaption>
+</figure>
+
+In my opinion, one of the the biggest challenges with Jira is also one of its strength: Jira offers a lot of possibilities for customization. Trust me, it’s very easy to fall in the trap of wanting to configure all the nitty gritty details of one team’s operation.
+Want to prevent an issue from moving to the next stage if testing hasn’t been documented ? Sure, no problem. Require 2 reviewers before a ticket can be completed ? Of course. And the list goes on.
+
+The challenge with a complex workflow is that it easily gets in the way of the development team, in particular around exceptions during burst period. For example on one of our project we required documentation instructions to be written for stories. It makes sense, right ? Now if we look back at the 6,000 issues we have on this particular project, only 175 were populated with actual documentation instructions, most of which were just links to pull requests. Not the most successful workflow rule !
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: For example,

Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: projects

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+GitHub on the other end, simply doesn’t allow for advanced workflow customizations. Issues can move freely between statuses, with no possibility to add validations steps at the issue level. This gives much more freedom to the team but also helps entrust all team members. It may take a few reminders at the beginning (please don’t forget to estimate unplanned urgent issues), but it’s definitely worthwhile.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: Do you mean: Github, on the other hand, ?

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+# Are GitHub issues the ideal solution ?
+
+GitHub provides features for [project management](https://github.com/features/project-management/) out of the box but we quickly realized those would not be sufficient for our needs.
+
+## Cross-organizations project boards
+
+Although you can use a GitHub project board across multiple repositories of the same organization, GitHub projects does not allow for cross-organization boards, which is a definite no-go in our context.
+
+We have one GitHub organization for [Overtuer](http://www.overture.bio/) and one organization per project. Project-specific implementation go into the project organization (for example: ICGC, Kids-First, HCMI), reusable components go into the Overture organization.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: Overture is mispelled

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+Each scrum team uses its own Agile scrum board, and the lack of cross-organizational project boards would definitely be a blocker (more on that later).
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/jira-github/github-overture.png" />
+    <figcaption>Some of Overture’s issues on GitHub projects</figcaption>
+</figure>
+
+## Story Points
+
+GitHub doesn’t have a dedicated field to store Story Points, neither provide metrics around those (velocity & burndown in particular). A workaround for points recording is to use Labels but there are no build-in tools for charting.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: would suggest "neither does it"

Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: should be built-in

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+The recommended approach when working with GitHub issues is as follow:
+
+* Epic: Regular issue attached to an “Epic” label, Stories linked in the issue description.
+* Story: Regular issue
+* SubTask: List in the description of the parent Story, with checkboxes ( - [ ] ).
+
+## Labels Management
+
+Labels are managed at the repository level, and although Labels can easily be created at any point in time in the interface, there is no provided feature to ensure their consistency across repositories and organizations.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: sometimes its frustrating to continually create the same label across reepositories as well. jira labels are global!

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+## I miss JQL though
+
+Jira has its own internal query language called [JQL](https://confluence.atlassian.com/jirasoftwarecloud/advanced-searching-764478330.html), sadly there is nothing as powerful available in GitHub ecosystem.
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/jira-github/issues-search.png" />
+    <figcaption>GitHub issues search</figcaption>
+</figure>
+
+# GitHub very extensive API
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: Githubs' ?

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+But GitHub played it very smart by providing 2 very extensive APIs ([v3 using REST](https://developer.github.com/v3/) , [v4 using GraphQL](https://developer.github.com/v4/)), their [associated libraries](https://developer.github.com/v3/libraries/) and a live instance of [GraphiQL](https://developer.github.com/v4/explorer/) making it extremely simple to extend GitHub functionality to cover particular use cases.
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/jira-github/graphql-explorer.png" />
+    <figcaption>How many issues in Arranger ?</figcaption>
+</figure>
+
+GitHub provides a strong foundation, and open the door to third party vendors to extend their core functionalities, encouraging an ecosystem of applications to grow alongside (we’ll see how this plays with Microsoft joining the game).
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: and opens the door

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+But does this call for a schema-less user-defined JSON info field directly attached to an issue node in GitHub database ? That would be a great improvement allowing third party apps to store most of their metadata there.
+
+# Conclusion
+At the moment there isn’t an ideal tool. We are ok (I’m not going to say happy, because of the issues above) with a mix between third party apps and native GitHub, but I’m looking at ways to bring more into core GitHub issues (through labels for example) to limit the above-mentioned long-run dependency.
+
+From a project manager point of view, I like the approach and simplicity of it (less is more). In a way, it’s simplicity brings it much closer to a traditional “post-it” Agile board. Most of the key metrics are there and it’s ease of access for our users makes it a great solution for the time being.
+
+We constantly adjust our Agile methodology & tools to best fit our needs, so with no doubts thing will change in the future, but for the time being this is an efficient and workable setup.
+
+# Conclusion bis, so GitHub or Jira for my project ?
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: is this a typo?

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+
+Both solutions are perfectly fine and mostly depends of your use case. If you need strict control in your workflow, Jira is probably your best bet. If you want flexibility while giving your team a sense of ownership towards their workflow, then GitHub issues is a great option.
```
Comment (rosibaj) [inline _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: mostly depends on your use case

**Review by rosibaj — COMMENTED** [review]
Overall this is really great!  Its pretty comprehensive on the comparison. Once more edit would be needed for spacing - I saw a few instances when reading on the PR that have spaces before punctuation (like this !).  

There are a couple points I also have found:

- Github does not let you *link* issues the same way.  This is actually a really useful feature to reference older tasks, or a bug related to something.  In JIRA it combines them as a list. In Github, itis added a note in the comments if you link it, which is hard to follow. 

- Github doesn't let you attach files other than images.  I think I get why as a storage issue, but man, this is irritating sometimes!!

- You mentioned this, but I would second that a huge advantage is that it links to PRs way better.  As a tester, it's easier to help identify issues since the code is *literally* right there with the user ticket!

File: _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md (hunk 9 above)
Comment (Fgerthoffert) [reply _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: sorry, which part are you referring to ?

Comment (rosibaj) [reply _posts/francois_gerthoffert/2018-10-02-Transitioning-away-from-Jira-Lessons-learnt.md]: "# Conclusion bis,"  <-- i just wasn't sure what "bis" means..  It might just be a phrase i don't know :)
